### PR TITLE
refactor: remaining `_comp_compgen_{help,usage}` changes

### PR DESCRIPTION
**Pull request by akinomyoga**
These are the remaining ones for the `_parse_{help,usage}` refactoring.

### CODE REVIEW

File: completions/dnssec-keygen [hunk 1]
```diff
@@ -37,8 +37,10 @@ _comp_cmd_dnssec_keygen()
     esac
 
     if [[ $cur == -* ]]; then
-        COMPREPLY=($(compgen -W '$(_parse_help "$1" | \
-            command sed -e "s/:\$//")' -- "$cur"))
+        _comp_compgen -R help
+        ((${#COMPREPLY[@]})) &&
+            _comp_compgen -- -W '$(printf "%s\n" "${COMPREPLY[@]}" |
+                command sed -e "s/:\$//")'
```
Comment (scop) [inline completions/dnssec-keygen:43]:
```suggestion
            _comp_compgen -- -W '"${COMPREPLY[@]%:}"'
```

File: completions/links [hunk 2]
```diff
@@ -80,8 +80,10 @@ _comp_cmd_links()
     esac
 
     if [[ $cur == -* ]]; then
-        COMPREPLY=($(compgen -W '$(_parse_help "$1" |
-            command grep -vF -- "->")' -- "$cur"))
+        _comp_compgen -R help
+        ((${#COMPREPLY[@]})) &&
+            _comp_compgen -- -W '$(printf "%s\n" "${COMPREPLY[@]}" |
+                command grep -vF -- "->")'
```
Comment (scop) [inline completions/links:86]:
```suggestion
            _comp_compgen -- -W '"${COMPREPLY[@]}"' -X "->"
```

File: completions/zopfli [hunk 3]
```diff
@@ -12,8 +12,10 @@ _comp_cmd_zopfli()
     esac
 
     if [[ $cur == -* ]]; then
-        COMPREPLY=($(compgen -W \
-            '$(_parse_help "$1" -h | command sed -e "s/#$//")' -- "$cur"))
+        _comp_compgen -R help -- -h
+        ((${#COMPREPLY[@]})) &&
+            _comp_compgen -- -W '$(printf "%s\n" "${COMPREPLY[@]}" |
+                command sed -e "s/#$//")' -- "$cur"
```
Comment (scop) [inline completions/zopfli:18]:
```suggestion
            _comp_compgen -- -W '"${COMPREPLY[@]%#}"'
```

**Review by scop — CHANGES_REQUESTED** [review]

File: completions/links (hunk 2 above)
Comment (akinomyoga) [reply completions/links:86]: Thank you for these suggestions! You are right, we can do them just by the builtin feature. I would like to apply these changes, but I'd like to confirm one thing here. The above `grep -vF -- "->"` is technically equivalent to `-X '*->*'`, but does the intent actually to remove the exact matching to `->`?

Comment (akinomyoga) [reply completions/links]: Thank you. I have applied the changes.

**Review by scop — APPROVED** [review]

**Pull request merged**